pkg/providers/tinkerbell/hardware/catalogue_hardware.go

```diff
@@ -1,21 +1,67 @@
 package hardware
 
 import (
-	"github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
+	"fmt"
+
+	tinkv1alpha1 "github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	eksav1alpha1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/constants"
 )
 
+type DiskExtractor struct {
```

Comment:
Can we get code documentation on _all_ exported types at a minimum. The docs should explain what the scope of responsibility is and any behavior expectations.